### PR TITLE
changes as discussed in pull request #75

### PR DESCRIPTION
**Pull request by lukric**
`gen_data` now uses either `msf_dict` (which is as before) or `msf_dict_mortality`.

changed some parts of `gen_data` to support `msf_dict_mortality`; also changed `gen_data`'s input variable name from "disease" to "dictionary"

replaced all pipe-parts

added `tidyselect` to  imports

### CODE REVIEW

**Comment by zkamvar** [comment]
Hi @lukric,

For future reference: 

1. there is no need to close a PR and create a new one every time changes are requested.
2. you should not make changes on the master branch... it makes updating your fork difficult.

File: R/msf_dict_mortality.R [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  # clean col names
+  colnames(dat_dict) <- epitrix::clean_labels(colnames(dat_dict))
+
+  # fill NA values with previous non-NA value, replace "." in codes and names
+  dat_dict <- tidyr::fill(dat_dict,
+                          tidyselect::everything(), .direction = "down")
+
+  # minor tidying, e.g.: create "CodeX" assignments
+  dat_dict <- dplyr::mutate(dat_dict,
+    choice_code = replace(choice_code, choice_code == ".", NA),
```
Comment (zkamvar) [inline R/msf_dict_mortality.R]: tidyverse operations like this are best replaced by base R equivalents in packages.

For instance, this should be

```r
dat_dict$choice_code[dat_dict$choice_code == "."] <- NA
dat_dict$choice_name[dat_dict$choice_name == "."] <- NA
```

File: R/msf_dict_mortality.R [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    choice_name = replace(choice_name, choice_name == ".", NA))
+  dat_dict <- dplyr::mutate(dat_dict,
+                            data_element_name = column_name,
```
Comment (zkamvar) [inline R/msf_dict_mortality.R]: What is `column_name`? This is the first instance it appears here.

File: R/msf_dict_mortality.R [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                  )
+
+
+  # clean future var names
+  # excel names (data element shortname)
+  # csv names (data_element_name)
+  dat_dict_wide$data_element_shortname <- epitrix::clean_labels(dat_dict_wide$data_element_shortname)
+  dat_dict_wide$data_element_name      <- epitrix::clean_labels(dat_dict_wide$data_element_name)
+
+  dat_dict_wide <- data.frame(dat_dict_wide)
```
Comment (zkamvar) [inline R/msf_dict_mortality.R]: be careful of the implicit `stringsAsFactors = TRUE` here

File: R/msf_dict_mortality.R [hunk 4]
```diff
@@ -0,0 +1,76 @@
+# function to load MSF data dictionary for mortality surveys
+
+#' @importFrom rio import
```
Comment (zkamvar) [inline R/msf_dict_mortality.R]: Thank you for listing the imports at the top

File: R/msf_dict_mortality.R [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  dat_dict <- tidyr::fill(dat_dict,
+                          tidyselect::everything(), .direction = "down")
+
+  # minor tidying, e.g.: create "CodeX" assignments
+  dat_dict <- dplyr::mutate(dat_dict,
+    choice_code = replace(choice_code, choice_code == ".", NA),
+    choice_name = replace(choice_name, choice_name == ".", NA))
+  dat_dict <- dplyr::mutate(dat_dict,
+                            data_element_name = column_name,
+                            data_element_shortname = column_name,
```
Comment (zkamvar) [inline R/msf_dict_mortality.R]: Come to think of it, I'm not sure what's happening here.... so what's happening here?

**Comment by lukric** [comment]
> Hi @lukric,
> 
> For future reference:
> 
> 1. there is no need to close a PR and create a new one every time changes are requested.
> 2. you should not make changes on the master branch... it makes updating your fork difficult.

hey @zkamvar 

thanks for the tips. just for my understanding: i do the pull request into the r4epi master branch?

File: R/msf_dict_mortality.R (hunk 2 above)
Comment (lukric) [reply R/msf_dict_mortality.R]: `column_name` is from the dictionary and (to my understanding) contains the name of the question in the survey.

File: R/msf_dict_mortality.R (hunk 3 above)
Comment (lukric) [reply R/msf_dict_mortality.R]: thanks! will take care of it

File: R/msf_dict_mortality.R (hunk 5 above)
Comment (lukric) [reply R/msf_dict_mortality.R]: I tried to make the `msf_dict_mortality` output similar to the `msf_dict` one. However, after your comment I checked again, and `data_element_name` and `data_element_shortname` are not needed, as we can define the  `varnames` in `gen_data` anyway. Will remove those duplications.

**Comment by lukric** [comment]
> 1. there was a difference between the number of variables in the mortality dictionary (53) and the number of columns in the mortality data set. Was this purposeful?

Thanky you very much. There were some columns created hard for outbreak data.

> 2. There was a lot of non-standard evaluation in the msf_dict_mortality() function. I'm a big fan of using dplyr for data analysis, [but it's a completely different ballgame used in packages](https://cran.r-project.org/web/packages/dplyr/vignettes/programming.html). It would be good to reduce this to a minimum. One thing to do: if you are referring to a column in your data, you can use the `.data` variable from dplyr to reference the column (e.g. `select(dat_dict, .data$data_element_shortname, .data$code)`).

removed what was easily doable. do you have additional suggestions for the remaining parts?

**Comment by zkamvar** [comment]
Hi @lukric, 

Thank you for adjusting the code. I believe everything should work properly now with no surprises. You might want to add documentation to the `gen_data()` function that your data set has a different variable column than the disease data sets and needs to be called like `gen_data("Mortaility", varnames = "column_name")`

Also, there are a bunch of survey questions that you insert missing data into at the end. Can you write short comments about what they are doing since things like `q53_cq4a` doesn't really make much sense?

**Comment by zkamvar** [comment]
Note: I've made some changes, so you will have to pull first

**Comment by lukric** [comment]
Hey @zkamvar ,
thanks for your input (again) and your small changes. Please see my latest commit.

More of the plausibility checks of the randomly generated data might be added while I work on the mortality survey (depending on what is needed there). However, this will be part of future pull requests.

**Comment by zkamvar** [comment]
Thank you again for responding to
The changes. I will merge this as soon as I get to my computer.

**Comment by zkamvar** [comment]
@lukric, I've merged it. Congratulations on your first contribution and thank you for your patience through the review. 

A comment about moving forward: Because you made your commits directly to the master branch, updating your fork will be a painful process. It's better to just delete your fork and then re-fork. When you need to add something new, check out a new branch (you can do this directly from RStudio) and work there (known as [github flow](https://guides.github.com/introduction/flow/)).

new-git-branch

(note: I'm also doing this so that I can re-use this image in the wiki).

**Comment by lukric** [comment]
@zkamvar yay, its done :D 

I will work in other branches from now on (actually already started). That I used my master here was by mistake - probably I was too tired when I closed the old pull request and started this one.

**Pull request merged**